File: source-code/Server/backend/src/main/java/com/iasdietideals24/backend/entities/OffertaInversa.java
package com.iasdietideals24.backend.entities;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalTime;

@EqualsAndHashCode(callSuper = true)
@Getter
@Setter
@NoArgsConstructor
@Entity(name = "offerta_inversa")
public class OffertaInversa extends OffertaDiVenditore {
    @ManyToOne
    @JoinColumn(name = "asta_inversa_id_asta", nullable = false)
    @NonNull
    private AstaInversa astaRiferimento;

    // AllArgsConstructor
    public OffertaInversa(@NonNull LocalDate dataInvio, @NonNull LocalTime oraInvio, @NonNull BigDecimal valore, @NonNull Venditore venditoreCollegato, @NonNull AstaInversa astaRiferimento) {
        super(dataInvio, oraInvio, valore, venditoreCollegato);

        this.astaRiferimento = astaRiferimento;
        astaRiferimento.addOffertaRicevuta(this);
    }

    @Override
    public String toString() {
        return "OffertaInversa(astaRiferimento=" + this.getAstaRiferimento().getIdAsta() + ") is a " + super.toString();
    }
}

File: DietiDeals24/app/src/main/java/com/iasdietideals24/dietideals24/model/offerta/OffertaInversa.kt
package com.iasdietideals24.dietideals24.model.offerta

import com.iasdietideals24.dietideals24.model.account.AccountVenditore
import com.iasdietideals24.dietideals24.model.asta.AstaInversa
import java.math.BigDecimal
import java.sql.Date
import java.sql.Time

class OffertaInversa(
    dataInvio: Date,
    oraInvio: Time,
    valore: BigDecimal,
    venditoreCollegato: AccountVenditore,

    private var astaRiferimento: AstaInversa
) : OffertaVenditore(dataInvio, oraInvio, valore, venditoreCollegato) {

    fun getAstaRiferimento(): AstaInversa {
        return astaRiferimento
    }

    fun setAstaRiferimento(astaRiferimento: AstaInversa) {
        this.astaRiferimento = astaRiferimento
    }
}